### PR TITLE
Update tp sharding annotation for MoE block

### PR DESCRIPTION
**Pull request by RissyRan**
# Description

Update tp sharding annotation for shard_map to match same sharding strategy in MaxText

# Test

* unit test passes
* xprof - [link](https://screenshot.googleplex.com/5BkmFGyVsxrH9MD)

### CODE REVIEW

File: MaxText/layers/linears.py [hunk 1]
```diff
@@ -221,7 +221,7 @@ def __call__(self, inputs, decode: bool = False, deterministic: bool = False):
           dtype=self.dtype,
           weight_dtype=self.weight_dtype,
           kernel_init=self.kernel_init,
-          kernel_axes=("embed", "num_activations", "mlp"),
```
Comment (gobbleturk) [inline MaxText/layers/linears.py:224]: nit: I would leave these as mlp just b/c the mix of strings and common_types looks strange, but either is fine...

File: MaxText/layers/linears.py [hunk 2]
```diff
@@ -385,27 +385,25 @@ def gmm(inputs, kernel, group_sizes):
         shard_map.shard_map,
         mesh=self.mesh,
         in_specs=(
-              (nn.logical_to_mesh_axes((BATCH, None, EMBED))),
               (nn.logical_to_mesh_axes((BATCH, None, None))),
```
Comment (gobbleturk) [inline MaxText/layers/linears.py:388]: Can we add a comment here (or at top of annotations) saying we purposefully AG the input activations over TP to follow a specific sharding strategy outlined in https://parsa.epfl.ch/course-info/cs723/papers/Megatron.pdf 

(we can include this link the comment)

**Review by gobbleturk — APPROVED** [review]

File: MaxText/layers/linears.py (hunk 1 above)
Comment (RissyRan) [reply MaxText/layers/linears.py:224]: lol, np, I am fine with either way. I could change it back without using common_type.

File: MaxText/layers/linears.py (hunk 2 above)
Comment (RissyRan) [reply MaxText/layers/linears.py]: Sure thing, added

**Pull request closed without merging**